### ISSUE TITLE
feature request : support for mindmap block

### ISSUE THREAD

**Issue by alwinlin23** [lynchjames/obsidian-mind-map#5]
a dedicated mindmap block like

<pre>
```markmap
```
</pre>

would also be a nice option?

**Comment by anamyk** [comment]
Yes, this is a feature I am looking for. VS Code markmap plugin has this, too.

**Comment by xpgo** [comment]
> 
> 
> Yes, this is a feature I am looking for. VS Code markmap plugin has this, too.

Thanks for the plugin.

I have tried to implement this feature based on your codes: [Block Render Plugin for Obsidian](https://github.com/xpgo/obsidian-block-render-plugin), the plguin is just for self-using. It works but there are some known issures I don't know how to fix:

1. the inital vertical position of the map is not at the center, though you can drag it normally.
2. there will be an error saying `Error: <g> attribute transform: Expected number, "translate(NaN,NaN) scale(N…".` everytime the map is initialized. (Users do not know about this untill the developing panel is opened.)

I am not good at JS, I think you may know how to solve them and may implement it to your plugin.

**Comment by wikilpmk** [comment]
👍  Will like this option to have embedded mindmap within a document
